### PR TITLE
odb: add command to check if design is routed

### PR DESCRIPTION
**Pull request by eder-matheus**

### CODE REVIEW

File: src/odb/src/db/odb.tcl [hunk 1]
```diff
@@ -648,3 +648,18 @@ proc set_max_layer { args } {
   set block [$chip getBlock]
   $block setMaxRoutingLayer $maxLayer
 }
+
+sta::define_cmd_args "design_is_routed" {}
+
+proc design_is_routed { args } {
+  sta::parse_key_args "set_max_layer" args keys {} flags {}
```
Comment (gadfort) [inline src/odb/src/db/odb.tcl:655]: design_is_routed

**Review by gadfort — COMMENTED** [review]
Shouldn't this have a set of CI tests?

**Comment by eder-matheus** [comment]
> Shouldn't this have a set of CI tests?

Yes, I'm working on it. I'm also testing it in the CI to check if I'm missing some check.

File: src/odb/src/db/dbBlock.cpp [hunk 2]
```diff
@@ -4097,6 +4097,18 @@ void dbBlock::preExttreeMergeRC(double max_cap, uint corner)
   }
 }
 
+bool dbBlock::designIsRouted()
+{
+  for (dbNet* net : getNets()) {
+    const int pin_count = net->getBTermCount() + net->getITerms().size();
+    if (pin_count > 1 && net->getSWires().empty() && net->getWire() == nullptr
+        && !net->isConnectedByAbutment()) {
```
Comment (gadfort) [inline src/odb/src/db/dbBlock.cpp:4105]: if the SWires is not empty, you should probably verify that they actually have routing in them.
Same for getWire (ensure it actually contains something).

Comment (eder-matheus) [reply src/odb/src/db/dbBlock.cpp]: I was able to simplify it by getting the wire count of each net. This function already cover both special and non-special nets.

**Comment by eder-matheus** [comment]
@maliberty This is ready for review.

**Review by maliberty — APPROVED** [review]

**Pull request merged**